src/Symfony/Bundle/FrameworkBundle/Command/RouterDebugCommand.php

```diff
@@ -115,4 +121,35 @@ private function findRouteNameContaining(string $name, RouteCollection $routes):
 
         return $foundRoutesNames;
     }
+
+    private function filterRoutes(RouteCollection $routes, InputInterface $input)
+    {
+        if (!empty($methods = $input->getOption('method'))) {
+            $routes = $routes->filter(function (Route $route) use ($methods) {
+                return \array_intersect($methods, $route->getMethods());
+            });
+        }
+
+        if (null !== ($scheme = $input->getOption('scheme'))) {
+            if (!\in_array($scheme, array('http', 'https'))) {
+                throw new InvalidArgumentException(\sprintf('The "%s" scheme is not valid. Valid schemes are "http" and "https".', $scheme));
+            }
+
+            $routes = $routes->filter(function (Route $route) use ($scheme) {
+                return \in_array(\strtolower($scheme), $route->getSchemes(), true);
+            });
+        }
+
+        if (null !== ($hostRegex = $input->getOption('match-host'))) {
+            try {
+                $routes = $routes->filter(function (Route $route) use ($hostRegex) {
+                    return (bool) \preg_match('/'.$hostRegex.'/', $route->getHost());
```

Comment:
The regex needs a `preg_quote` for safety. Even if an host is not supposed to contain a slash IMHO.

Reply:
Why adding `preg_quote` here ? 

If I do that, every regex are now "valid" but does not match any host. If I want to filter with host ending by `example.com`, I'll need to pass a regex like this `example\.com$`, with `preg_quote` this will be transformed to `example\\\.com\$` and will match anything.

Is there something I am missing here ?

Reply:
Sorry I reread my first comment and it was written badly.

`preg_quote` is useful so you don't have to do the escaping yourself on special regular expression characters + on the delimiter (because you cannot expect that the users are gonna check the command source to see what delimiter it uses). My first comment was about the fact that even if the delimiter (`/`) should not be in a host, I would still add it for safety.

However my suggestion is not compatible with your example (if you want to check if an host only begin or end with a pattern (since the `^` and `$` would be escaped as well)). I just thought initially that the filtering on the host was only a "contains".

Reply:
Ok, so I will let as is.

But you are right, I need to find a way to inform the user that the delimiter I use is `/`. Maybe in the description of the option ? Or do you have any suggestion ?